Guard the SIGHUP remap reload with WITH_REGEX.

The 6.0 rewrite of the standalone signal handling (caught_sighup ->
reload_signal) lost the `#ifdef WITH_REGEX` that used to wrap this block, so
`./configure --disable-remap` -- or any host without <regex.h> -- fails to
build tftpd:

    tftpd.c:826:17: error: 'rewrite_file' undeclared (first use in this function)
    tftpd.c:827:27: error: 'rewrite_rules' undeclared (first use in this function)

--- a/tftpd/tftpd.c
+++ b/tftpd/tftpd.c
@@ -823,10 +823,12 @@
 
         if (reload_signal) {
             reload_signal = 0;
+#ifdef WITH_REGEX
             if (rewrite_file) {
                 freerules(rewrite_rules);
                 rewrite_rules = read_remap_rules(rewrite_file);
             }
+#endif
         }
 
         rv = pollset_poll(listen_set, POLLSET_IN, waittime);
